Mathlib/Data/List/FinRange.lean

```diff
@@ -67,6 +72,18 @@ theorem nodup_ofFn {α n} {f : Fin n → α} : Nodup (ofFn f) ↔ Function.Injec
     exact h.imp_right ih
 #align list.nodup_of_fn List.nodup_ofFn
 
+/-- Folding over a list obtained from a function `g : Fin n → α` is equivalent to
+    folding over the list of indices `0, 1, ..., n-1` and invoking `g` for each index -/
+theorem foldl_ofFn {α β n} (f : α → β → α) (init : α) (g : Fin n → β) :
+    (List.ofFn g).foldl f init = (List.finRange n).foldl (fun a i => f a (g i)) init := by
+  rw [ofFn_eq_map, foldl_map]
+
+/-- Folding over a list obtained from a function `g : Fin n → α` is equivalent to
+    folding over the list of indices `0, 1, ..., n-1` and invoking `g` for each index -/
+theorem foldr_ofFn {α β n} (f : β → α → α) (init : α) (g : Fin n → β) :
+    (List.ofFn g).foldr f init = (List.finRange n).foldr (fun i a => f (g i) a) init := by
+  rw [ofFn_eq_map, foldr_map]
```

Comment:
With these new proofs in mind, it's not really clear to me that we want these lemmas at all; I think it makes more sense for the user to go via `ofFn_eq_map`, because then the whole API for `map` is at their fingertips. The trail you're blazing ends in us repeating the entire `map` API for `ofFn`!

Maybe the documentation for `ofFn` should more prominently display `ofFn_eq_map`.

Reply:
Fair point.

> Maybe the documentation for ofFn should more prominently display ofFn_eq_map.
This is slightly complicated, since `List.ofFn` is defined in Std, while `ofFn_eq_map` is from Mathlib.

In any case, now that I know about `ofFn_eq_map`, I can just use these simpler proofs in #5920

Reply:
`ofFn_eq_map` could probably be upstreamed to Std

Reply:
`finRange` is indeed also only in mathlib. The definition of `finRange` depends only on stuff in std, so we could upstream `ofFn_eq_map` and `finRange` together, if desired